src/solvers/smt/smt_overflow.cpp

```diff
@@ -204,12 +204,20 @@ smt_astt smt_convt::overflow_neg(const expr2tc &expr)
   if (int_encoding)
     return mk_smt_bool(false);
 
-  // Single failure mode: MIN_INT can't be neg'd
+  // Extract operand
   const overflow_neg2t &neg = to_overflow_neg2t(expr);
   unsigned int width = neg.operand->type->get_width();
 
-  expr2tc min_int =
-    constant_int2tc(neg.operand->type, -BigInt::power2(width - 1));
-  expr2tc val = equality2tc(neg.operand, min_int);
+  // Ensure operand is an integer (convert boolean to signed integer if needed)
+  expr2tc operand = is_bool_type(neg.operand->type)
```

Comment:
Can't you always do the typecast? Our simplifier should be smart enough to handle double typecast.

Reply:
Implemented as suggested.